### PR TITLE
bpo-1635741: convert an  _lsprof method to argument clinic

### PR DESCRIPTION
**Pull request by koubaa**
https://bugs.python.org/issue1635741

### CODE REVIEW

**Comment by koubaa** [comment]
@vstinner Please review, this will be the first stage from https://github.com/python/cpython/pull/22220

File: Modules/_lsprof.c [hunk 1]
```diff
@@ -556,49 +563,56 @@ static int statsForEntry(rotating_node_t *node, void *arg)
...
-    reccallcount  how many times this is called recursively\n\
-    totaltime     total time spent in this call\n\
-    inlinetime    inline time (not in further subcalls)\n\
-");
+/*[clinic input]
+_lsprof.Profiler.getstats
 
-static PyObject*
-profiler_getstats(ProfilerObject *pObj, PyObject* noarg)
+    cls: defining_class
```
Comment (vstinner) [inline Modules/_lsprof.c]: Can you please remove it from this PR?

File: Modules/_lsprof.c [hunk 2]
```diff
@@ -556,49 +563,56 @@ static int statsForEntry(rotating_node_t *node, void *arg)
...
+    inlinetime    inline time (not in further subcalls)
+[clinic start generated code]*/
+
+static PyObject *
+_lsprof_Profiler_getstats_impl(ProfilerObject *self, PyTypeObject *cls)
+/*[clinic end generated code: output=1806ef720019ee03 input=445e193ef4522902]*/
 {
     statscollector_t collect;
-    if (pending_exception(pObj)) {
+    if (pending_exception(self)) {
```
Comment (vstinner) [inline Modules/_lsprof.c:599]: Is it possible to rename the self parameter to pObj to avoid these unneeded changes?

Comment (koubaa) [reply Modules/_lsprof.c:599]: @vstinner I don't know if clinic supports that, I couldn't find this in the documentation.

Comment (vstinner) [reply Modules/_lsprof.c:599]: It works using this syntax:

```
_lsprof.Profiler.getstats

    self as pObj: self

```

Comment (koubaa) [reply Modules/_lsprof.c:599]: @vstinner I tried this, but it leads to this diff (the argument type changes):
```
-_lsprof_Profiler_getstats_impl(ProfilerObject *self)
-/*[clinic end generated code: output=9461b451e9ef0f24 input=ade04fa384ce450a]*/
+_lsprof_Profiler_getstats_impl(PyObject *pObj)
+/*[clinic end generated code: output=3e650300ffbc0716 input=91d29896c19da4ce]*/
```

File: Misc/NEWS.d/next/Core and Builtins/2020-09-14-09-15-59.bpo-1635741.UnXTND.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Convert the getstats method of the :mod:`_lsprof` extension module to
+argument clinic
```
Comment (vstinner) [inline Misc/NEWS.d/next/Core and Builtins/2020-09-14-09-15-59.bpo-1635741.UnXTND.rst]: Please remove the NEWS entry, it's not worth it.

File: Modules/_lsprof.c (hunk 2 above)
Comment (vstinner) [reply Modules/_lsprof.c:599]: Oh, I didn't notice. I give up with Argument Clinic, "self" is an explicit name, ignore my comment.

**Review by vstinner — APPROVED** [review]
LGTM but please remove the NEWS entry.

File: Misc/NEWS.d/next/Core and Builtins/2020-09-14-09-15-59.bpo-1635741.UnXTND.rst (hunk 3 above)
Comment (koubaa) [reply Misc/NEWS.d/next/Core and Builtins/2020-09-14-09-15-59.bpo-1635741.UnXTND.rst]: @vstinner done

**Pull request merged**